### PR TITLE
Add new docker image for pinot servicemanager

### PR DESCRIPTION
**Pull request by ravisingal**

### CODE REVIEW

File: pinot-servicemanager/etc/pinot-controller.conf [hunk 1]
```diff
@@ -0,0 +1,7 @@
+controller.helix.cluster.name=hypertrace-views
+controller.port=9000
+controller.data.dir=/var/pinot/controller/data
+# TODO: see if this can be CLI arg or ENV
+controller.zk.str=kafka-zookeeper:2181
```
Comment (kotharironak) [inline pinot-servicemanager/etc/pinot-controller.conf:5]: Yes for todo, we should explore if we can pass this via arg.

File: pinot-servicemanager/etc/pinot-controller.conf [hunk 2]
```diff
@@ -0,0 +1,7 @@
+controller.helix.cluster.name=hypertrace-views
```
Comment (kotharironak) [inline pinot-servicemanager/etc/pinot-controller.conf:1]: This one also if we can remove and able to pass via arg.

File: pinot-servicemanager/build.gradle.kts [hunk 3]
```diff
@@ -0,0 +1,11 @@
+plugins {
+  id("org.hypertrace.docker-publish-plugin")
+}
+
+hypertraceDocker {
+  defaultImage {
+    imageName.set("pinot")
+    tags.forEach { it.onlyIf { false } }
+    tag("servicemanager")
```
Comment (aaron-steinfeld) [inline pinot-servicemanager/build.gradle.kts:9]: `hypertrace/pinot:servicemanager` ? That would result in the image having no versioning and sharing a name with an image built completely differently. Would it make more sense to give it its own name, similar to the kafka-zookeeper image?

File: pinot-servicemanager/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+ENV JAVA_OPTS="-Xms128M -Xmx192M -XX:MaxDirectMemorySize=64M -XX:+ExitOnOutOfMemoryError"
+
+WORKDIR ${PINOT_HOME}
+
+# Setup pinot user so that the image doesn't run as root
+RUN adduser -g '' -h ${PINOT_HOME} -D pinot && \
+    mkdir -p ${PINOT_VAR}/server/data ${PINOT_VAR}/controller/data && \
+    chown -R pinot ${PINOT_VAR}
+
+USER pinot
```
Comment (buchi-busireddy) [inline pinot-servicemanager/Dockerfile:29]: Declare the `USER` above and use it in all the commands.

File: pinot-servicemanager/build.gradle.kts (hunk 3 above)
Comment (ravisingal) [reply pinot-servicemanager/build.gradle.kts]: done.

**Comment by aaron-steinfeld** [comment]
Gradle side changes are fine. Will let someone with more pinot context approve the actual dockerfile.

**Comment by ravisingal** [comment]
> Gradle side changes are fine. Will let someone with more pinot context approve the actual dockerfile.

The image out of this dockerfile is being tested in docker-compose.

File: pinot-servicemanager/Dockerfile (hunk 4 above)
Comment (ravisingal) [reply pinot-servicemanager/Dockerfile]: done.

Comment (ravisingal) [reply pinot-servicemanager/Dockerfile]: @buchi-busireddy circleci builds are failing after this change:
```
Step 15/19 : COPY --from=pinot --chown=${USER} /pinot ${PINOT_HOME}
unable to convert uid/gid chown string to host mapping: can't find uid for user ${USER}: no such user: ${USER}
```
circleci uses docker version `17.09.0-ce` in their build environment while this is fixed in docker version `19.0.3.0`.
https://github.com/moby/moby/issues/35018

**Pull request merged**